mbtiles/src/queries.rs

```diff
@@ -409,8 +394,10 @@ FROM tiles;"
     .fetch_one(conn)
     .await?;
 
-    let min_zoom = validate_zoom(info.min_zoom, "zoom_level")?;
-    let max_zoom = validate_zoom(info.max_zoom, "zoom_level")?;
+    #[allow(clippy::cast_possible_truncation)]
+    let min_zoom = validate_zoom(info.min_zoom.map(|v| v as i32), "zoom_level")?;
+    #[allow(clippy::cast_possible_truncation)]
+    let max_zoom = validate_zoom(info.max_zoom.map(|v| v as i32), "zoom_level")?;
```

Comment:
I don't think the truncation here is the best option.

`fn validate_zoom` does check if the value can be casted to `u8` via `u8::try_from`

Changing to `fn validate_zoom(zoom: Option<i64>, zoom_name: &'static str) -> MbtResult<Option<u8>>` seems like a better option

```suggestion
    let min_zoom = validate_zoom(info.min_zoom, "zoom_level")?;
    let max_zoom = validate_zoom(info.max_zoom, "zoom_level")?;
```